wgsl/index.bs

```diff
@@ -9133,6 +9133,170 @@ Cube faces are square, so the x and y components of the result are equal.
 If `level` is outside the range `[0, textureNumLevels(t))` then any valid value
 for the return type may be returned.
 
+### `textureGather` and `textureGatherComponent` ### {#texturegather}
+
+A <dfn noexport>texture gather</dfn> operation reads from a 2D, 2D array, cube, or cube array texture,
+computing a four-element vector as follows:
+* Find the four texels that would be used in a sampling operation with linear filtering,
+    from [=mip level=] 0:
+    * Use the specified coordinate, array index (when present), and offset (when present).
+    * The texels are adjacent, forming a square, when considering their texture space coordinates (*u*,*v*).
+    * Selected texels at the texture edge, cube face edge, or cube corners are handled
+        as in ordinary texture sampling.
+* For each texel, read one channel and convert it into a scalar value.
+    * For non-depth textures, a zero-based `component` parameter specifies the channel to use.
+        * If the texture format supports the specified channel, i.e. has more than `component` channels:
+             * Yield scalar value `v[component]` when the texel value is `v`.
+        * Otherwise:
+             * Yield 0.0 when `component` is 1 or 2.
+             * Yield 1.0 when `component` is 3 (the alpha channel).
+    * For depth textures, yield the texel value. (Depth textures only have one channel.)
+* Yield the four-element vector, arranging scalars produced by the previous step into components
+    according to the relative coordinates of the texels, as follows:
+    * <table>
+        <thead class='data'><td>Result component<td>Relative texel coordinate</thead>
+        <tr><td>x<td>(*u*<sub>min</sub>,*v*<sub>max</sub>)
+        <tr><td>y<td>(*u*<sub>max</sub>,*v*<sub>max</sub>)
+        <tr><td>z<td>(*u*<sub>max</sub>,*v*<sub>min</sub>)
+        <tr><td>w<td>(*u*<sub>min</sub>,*v*<sub>min</sub>)
+        </table>
+
+TODO: The four texels are the "sample footprint" that should be described by the WebGPU spec.
+https://github.com/gpuweb/gpuweb/issues/2343
+
+```rust
+textureGather(t: texture_2d<T>, s: sampler, coords: vec2<f32>) -> vec4<T>
```

Comment:
are all gather operations today and tomorrow going to only concern themselves with mip level 0?

Reply:
I think yes.  All the APIs only support mip level 0.
To support higher mip levels, use a GPUTextureView with a higher mipLevel.